File: ecobertura.core/test/ecobertura/core/log/LogStatusTest.java
package ecobertura.core.log;

import static org.junit.Assert.*;


import org.eclipse.core.runtime.Status;
import org.junit.Test;

import ecobertura.core.log.LogStatus;

public class LogStatusTest {

	private LogStatus status;
	
	@Test
	public void testFromMessageWithSeverityInfo() {
		status = LogStatus.fromMessageWithSeverity("hello", Status.INFO);
		assertLogStatusMessage("hello", Status.INFO);
	}

	@Test
	public void testFromMessageWithSeverityWarning() {
		status = LogStatus.fromMessageWithSeverity("", Status.WARNING);
		assertLogStatusMessage("", Status.WARNING);
	}
	
	@Test
	public void testFromMessageWithSeverityError() {
		status = LogStatus.fromMessageWithSeverity("bla hello", Status.ERROR);
		assertLogStatusMessage("bla hello", Status.ERROR);
	}
	
	private void assertLogStatusMessage(String message, int severity) {
		assertEquals(message, status.getMessage());
		assertEquals(severity, status.getSeverity());
		assertNull(status.getException());
	}
	
	@Test(expected=AssertionError.class)
	public void testFromMessageWithSeverityNullMessage() {
		LogStatus.fromMessageWithSeverity(null, Status.INFO);
	}
		
	@Test(expected=AssertionError.class)
	public void testFromMessageWithInvalidSeverity() {
		LogStatus.fromMessageWithSeverity("bla", Status.CANCEL);
	}
		
	@Test
	public void testFromExceptionWithSeverityInfo() {
		status = LogStatus.fromExceptionWithSeverity("hello", new Exception("hello"), Status.INFO);
		assertLogStatusException("hello", new Exception("hello"), Status.INFO);
	}

	@Test
	public void testFromExceptionWithSeverityWarning() {
		status = LogStatus.fromExceptionWithSeverity("", new Exception(""), Status.WARNING);
		assertLogStatusException("", new Exception(""), Status.WARNING);
	}
	
	@Test
	public void testFromExceptionWithSeverityError() {
		status = LogStatus.fromExceptionWithSeverity("bla bla", new Throwable("blurp"), Status.ERROR);
		assertLogStatusException("bla bla", new Throwable("blurp"), Status.ERROR);
	}
	
	private void assertLogStatusException(String message, Throwable throwable, int severity) {
		assertEquals(message, status.getMessage());
		assertEquals(throwable.getClass(), status.getException().getClass());
		assertEquals(throwable.getMessage(), status.getException().getMessage());
		assertEquals(severity, status.getSeverity());
	}
	
	
	@Test(expected=AssertionError.class)
	public void testFromExceptionWithSeverityNullMessage() {
		LogStatus.fromExceptionWithSeverity(null, new Throwable("bla"), Status.INFO);
	}
		
	@Test(expected=AssertionError.class)
	public void testFromExceptionWithSeverityNullThrowable() {
		LogStatus.fromExceptionWithSeverity("bla", null, Status.INFO);
	}
		
	@Test(expected=AssertionError.class)
	public void testFromExceptionWithInvalidSeverity() {
		LogStatus.fromExceptionWithSeverity("bla", new Throwable("bla"), Status.CANCEL);
	}
		
}

File: ecobertura.core/src/test/scala/ecobertura/core/log/LogStatusTest.scala
/*
 * This file is part of eCobertura.
 * 
 * Copyright (c) 2009, 2010 Joachim Hofer
 * All rights reserved.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 */
package ecobertura.core.log

import org.junit.Assert._
import org.junit.Test;

import org.eclipse.core.runtime.Status

import _root_.ecobertura.core.log.LogStatus._

class LogStatusTest {
  var status: Status = null

  @Test
  def testFromMessageWithSeverityInfo = {
    status = fromMessageWithSeverity("hello", Severity.Info)
    assertLogStatusMessage("hello", Severity.Info)
  }

  @Test
  def testFromMessageWithSeverityWarning = {
    status = fromMessageWithSeverity("", Severity.Warning)
    assertLogStatusMessage("", Severity.Warning)
  }
  
  @Test
  def testFromMessageWithSeverityError = {
    status = fromMessageWithSeverity("bla hello", Severity.Error)
    assertLogStatusMessage("bla hello", Severity.Error)
  }
  
  private def assertLogStatusMessage(message: String, severity: Severity.Value) = {
    assertEquals(message, status.getMessage)
    assertEquals(severity.id, status.getSeverity)
    assertNull(status.getException)
  }
  
  @Test(expected=classOf[AssertionError])
  def testFromMessageWithSeverityNullMessage : Unit = fromMessageWithSeverity(null, Severity.Info)
  
  @Test
  def testFromExceptionWithSeverityInfo = {
    status = fromExceptionWithSeverity("hello", new Exception("hello"), Severity.Info)
    assertLogStatusException("hello", new Exception("hello"), Severity.Info)
  }
  
  @Test
  def testFromExceptionWithSeverityWarning = {
    status = fromExceptionWithSeverity("", new Exception(""), Severity.Warning)
    assertLogStatusException("", new Exception(""), Severity.Warning)
  }
  
  @Test
  def testFromExceptionWithSeverityError = {
    status = fromExceptionWithSeverity("bla bla", new Throwable("blurp"), Severity.Error)
    assertLogStatusException("bla bla", new Throwable("blurp"), Severity.Error)
  }
  
  @Test(expected=classOf[AssertionError])
  def testFromExceptionWithSeverityNullMessage : Unit = 
      fromExceptionWithSeverity(null, new Throwable("bla"), Severity.Info)
  
  @Test(expected=classOf[AssertionError])
  def testFromExceptionWithSeverityNullThrowable : Unit =
      fromExceptionWithSeverity("bla", null, Severity.Info);

  private def assertLogStatusException(
      message: String, throwable: Throwable, severity: Severity.Value) = {
    assertEquals(message, status.getMessage)
    assertEquals(throwable.getClass, status.getException.getClass)
    assertEquals(throwable.getMessage, status.getException.getMessage)
    assertEquals(severity.id, status.getSeverity)
  }
}
